src/frontend/src/lib/components/dapps/DappModal.svelte

```diff
@@ -0,0 +1,84 @@
+<script lang="ts">
+	import { Html, Modal } from '@dfinity/gix-components';
+	import { nonNullish } from '@dfinity/utils';
+	import DappTags from '$lib/components/dapps/DappTags.svelte';
+	import IconGitHub from '$lib/components/icons/IconGitHub.svelte';
+	import IconTwitter from '$lib/components/icons/IconTwitter.svelte';
+	import ExternalLink from '$lib/components/ui/ExternalLink.svelte';
+	import ExternalLinkIcon from '$lib/components/ui/ExternalLinkIcon.svelte';
+	import Img from '$lib/components/ui/Img.svelte';
+	import ImgBanner from '$lib/components/ui/ImgBanner.svelte';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { modalStore } from '$lib/stores/modal.store';
+	import type { DappDescription } from '$lib/types/dappDescription';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+
+	export let dAppDescription: DappDescription;
+	const formattedUrl = new URL(dAppDescription.website).hostname.replace('www.', '');
+</script>
+
+<Modal on:nnsClose={modalStore.close}>
+	<div slot="title">
+		<span class="text-center text-xl">{dAppDescription.name}</span>
+	</div>
+
+	<div class="stretch flex flex-col gap-4 pt-4">
+		{#if nonNullish(dAppDescription.screenshots) && dAppDescription.screenshots.length > 0}
```

Comment:
Isn't this the same piece of code as the one we reviewed recently? Shouldn't you create a component for it?

Reply:
The promobanner expects a screenshot. This modal can display dapps without screenshots. The promobanner needs position absolute etc. I think the abstraction with the ImgBanner is good enough.

Reply:
Oki doki